src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -997,21 +985,25 @@ def __write_chunk__(cls, dset, data):
         """
         try:
             chunk_i = next(data)
-            # Determine the minimum array dimensions to fit the chunk selection
-            max_bounds = cls.__selection_max_bounds__(chunk_i.selection)
-            if not hasattr(max_bounds, '__len__'):
-                max_bounds = (max_bounds,)
-            # Determine if we need to expand any of the data dimensions
-            expand_dims = [i for i, v in enumerate(max_bounds) if v is not None and v > dset.shape[i]]
-            # Expand the dataset if needed
-            if len(expand_dims) > 0:
-                new_shape = np.asarray(dset.shape)
-                new_shape[expand_dims] = np.asarray(max_bounds)[expand_dims]
-                dset.resize(new_shape)
-            # Process and write the data
-            dset[chunk_i.selection] = chunk_i.data
         except StopIteration:
             return False
+        try:
+            # Determine the minimum array dimensions to fit the chunk selection
+            max_bounds = tuple([x.stop or 0 if isinstance(x, slice) else x+1 for x in chunk_i.selection])
+        except Exception as exc:
+            if isinstance(chunk_i.selection, int):
+                max_bounds = (chunk_i.selection+1, )
+            elif isinstance(chunk_i.selection, slice):
+                max_bounds = (chunk_i.selection.stop or 0, )
+            else:
+                msg = "Chunk selection %s must be a tuple of slices and/or integers" % str(chunk_i.selection)
+                raise TypeError(msg) from exc
```

Comment:
```suggestion
        if isinstance(chunk_i.selection, tuple):
            # Determine the minimum array dimensions to fit the chunk selection
            max_bounds = tuple([x.stop or 0 if isinstance(x, slice) else x+1 for x in chunk_i.selection])
        elif isinstance(chunk_i.selection, int):
             max_bounds = (chunk_i.selection+1, )
        elif isinstance(chunk_i.selection, slice):
              max_bounds = (chunk_i.selection.stop or 0, )
        else:
             msg = "Chunk selection %s must be a single int, single slice, or tuple of slices and/or integers" % str(chunk_i.selection)
             raise TypeError(msg) from exc
```

I.e,. Rather than using try/except I think this should check for the type explicitly. The reason for this are: 
- If ``chunk_i.selection`` is a list the try/except block will not necessarily always trigger an exception, which I believe could lead to unexpected behavior 
- Checking for all supported cases is more explicit and safer as the reason for the Exception may not be clear

I also updated the TypeError message to reflect that single ints and slices are permitted as selections as well.